### ISSUE TITLE
Add typed wallet support for Aptos and Sui

### ISSUE THREAD

**Issue by JoeHowarth** [wormhole-foundation/relayer-engine#23]
Parity with EVM and solana typed wallet support

Should be done after #22 
```[tasklist]
### Tasks
- [ ] https://github.com/wormhole-foundation/relayer-engine/issues/197
```

**Comment by Alivers** [comment]
Hello, Joe, I'm trying run Aptos<->Eth replayers, but the startup of the replayer exited with the below error. Is there any steps to skip the validateCommon Env？Thanks a lot~
```bash
relayer_engine is logging to the console at level [info]
2023-03-10 23:25:24.432|info|main: Messenger relayer plugin loaded  
Validating envs...
/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/config/validateConfig.ts:103
    throw new Error(
          ^
Error: Unrecognized chain 22 aptos
    at validateChainConfig (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/config/validateConfig.ts:103:11)
    at Array.map (<anonymous>)
    at validateCommonEnv (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/config/validateConfig.ts:58:7)
    at validateEnvs (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/config/index.ts:165:32)
    at readAndValidateEnv (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/index.ts:120:15)
    at Object.run (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/index.ts:55:9)
    at main (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/src/relayer.ts:49:23)
    at Object.<anonymous> (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/src/relayer.ts:70:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module.m._compile (/Users/aliver/.nvm/versions/node/v18.12.1/lib/node_modules/ts-node/src/index.ts:1618:23)
```

**Comment by JoeHowarth** [comment]
Thanks for raising this @Alivers, the pr above should fix the specific exception you hit, but we still don't support aptos or sui as first class yet. If you run into more problems please let us know and we'll try to fix them

**Comment by Alivers** [comment]
Thanks for your work. I had tried latest engine. These still are some problems existing. `runPluginSpyListener -> transformEmitterFilter` failed, Aptos not supported currently. Below: 

```bash
2023-03-11 18:08:35.703|error|spyEventSource: spy service threw an exception: Error: Encoding emitter address not implemented for wormhole chainId 22
    at encodeEmitterAddress (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:83:9)
    at transformEmitterFilter (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:51:27)
    at /Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:70:56
    at Array.map (<anonymous>)
    at runPluginSpyListener (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:68:20)
    at /Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:41:7
    at Array.forEach (<anonymous>)
    at createSpyEventSource (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:36:11)
    at Object.run (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:27:25)
    at Object.run (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/index.ts:81:25) {
  [stack]: 'Error: Encoding emitter address not implemented for wormhole chainId 22\n' +
    '    at encodeEmitterAddress (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:83:9)\n' +
    '    at transformEmitterFilter (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:51:27)\n' +
    '    at /Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:70:56\n' +
    '    at Array.map (<anonymous>)\n' +
    '    at runPluginSpyListener (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:68:20)\n' +
    '    at /Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:41:7\n' +
    '    at Array.forEach (<anonymous>)\n' +
    '    at createSpyEventSource (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/spyEventSource.ts:36:11)\n' +
    '    at Object.run (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/listener/listenerHarness.ts:27:25)\n' +
    '    at Object.run (/Users/aliver/workspace/repo/wormhole/xdapp-book/projects/messenger/node_modules/relayer-engine/relayer-engine/src/index.ts:81:25)',
  [message]: 'Encoding emitter address not implemented for wormhole chainId 22'
}  
```

**Comment by gabzim** [comment]
`Sui` is Supported, could we add `Aptos` support in a separate ticket and close this one?

**Comment by gabzim** [comment]
Created [#197](https://github.com/wormhole-foundation/relayer-engine/issues/197) to tackle Aptos support, sui is supported by doing `ctx.wallets.onSui(wallet => {...})`
